src/core/Table/hooks/useResizeColumns.tsx

```diff
@@ -0,0 +1,440 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+/**
+ * Copied from react-table as useResizerColumns does not have ownerDocument check also
+ * made some changes to resize sibling/next column on resize.
+ */
+/**
+ * MIT License
+ *
+ * Copyright (c) 2016 Tanner Linsley
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import React from 'react';
+import {
+  actions,
+  ActionType,
+  ColumnInstance,
+  HeaderGroup,
+  Hooks,
+  makePropGetter,
+  TableInstance,
+  TableKeyedProps,
+  TableState,
+  useGetLatest,
+} from 'react-table';
+
+export const useResizeColumns = <T extends Record<string, unknown>>(
+  ownerDocument: Document | undefined,
+) => (hooks: Hooks<T>) => {
+  hooks.getResizerProps = [defaultGetResizerProps(ownerDocument)];
+  hooks.stateReducers.push(reducer);
+  hooks.useInstanceBeforeDimensions.push(useInstanceBeforeDimensions);
+};
+
+const defaultGetResizerProps = (ownerDocument: Document | undefined) => (
+  props: TableKeyedProps,
+  {
+    instance,
+    header,
+    nextHeader,
+  }: {
+    instance: TableInstance;
+    header: HeaderGroup;
+    nextHeader: HeaderGroup;
+  },
+) => {
+  if (!ownerDocument) {
+    return props;
+  }
+
+  const { dispatch, flatHeaders } = instance;
+
+  const onResizeStart = (
+    e: React.TouchEvent & React.MouseEvent,
+    header: HeaderGroup,
+  ) => {
+    let isTouchEvent = false;
+    if (e.type === 'touchstart') {
+      // lets not respond to multiple touches (e.g. 2 or 3 fingers)
+      if (e.touches && e.touches.length > 1) {
```

Comment:
And then you can have just single if here `if (isTouchEvent && e.touches && e.touches.length) {return;}`

Reply:
Changed